src/test/java/bio/terra/service/filedata/azure/AzureSynapsePdaoSnapshotConnectedTest.java

```diff
@@ -294,6 +294,33 @@ public void testSnapshotByQuery() throws SQLException, IOException {
     createSnapshotRowIdParquet(snapshotByQueryTableRowCounts);
   }
 
+  @Test
+  public void testQuerySnapshotWithUnderscoreInFlightId() throws SQLException, IOException {
+    String ingestFlightId = "_" + UUID.randomUUID().toString();
+    SnapshotTable participantSnapshotTable = setupParticipantTable(ingestFlightId);
+    List<SnapshotTable> tables = List.of(participantSnapshotTable);
+    snapshot.snapshotTables(tables);
+    Map<String, Long> tableRowCounts =
+        azureSynapsePdao.createSnapshotParquetFiles(
+            tables, snapshotId, sourceDatasetDataSourceName, snapshotDataSourceName, false, null);
+    synapseUtils.addTableName(IngestUtils.formatSnapshotTableName(snapshotId, "participant"));
+
+    String snapshotParquetFileName =
+        IngestUtils.getSnapshotParquetFilePathForQuery(participantTable.getName());
+    synapseUtils.addParquetFileName(snapshotParquetFileName, snapshotStorageAccountResource);
+    List<String> snapshotFirstNames =
+        synapseUtils.readParquetFileStringColumn(
+            snapshotParquetFileName, snapshotDataSourceName, "first_name", true);
+    assertThat(
+        "List of names in snapshot should equal the dataset names",
+        snapshotFirstNames,
+        equalTo(List.of("Sally", "Bobby", "Freddy", "Charles")));
+    assertThat(
+        "Table row count should equal 4 for destination table",
+        tableRowCounts.get(participantTable.getName()),
+        equalTo(4L));
```

Comment:
Hamcrest has a map entry matcher that you can use here.
```suggestion
    assertThat(
        "Table row count should equal 4 for destination table",
        tableRowCounts,
        hasEntry(participantTable.getName(), 4));
```